src/effect.h

```diff
@@ -34,6 +35,59 @@ enum effect_rating {
     e_mixed     // The effect has good and bad parts to the one who has it.
 };
 
+struct caused_effect {
+    public:
+        efftype_id type;
+        /** Minimum parent effect intensity to apply the new effect. */
+        int intensity_requirement = 0;
+        /** If false, prevents application if the trigger was parent decaying to 0 duration. */
+        bool allow_on_decay = true;
+        /** If false, prevents application if the trigger was parent being removed with at least 1 turn of duration left. */
+        bool allow_on_remove = false;
+
+        /**
+         * Duration of the new effect.
+         * If 0, type's max duration will be used instead.
+         * Should be left at 0 for permanent effects.
+         */
+        time_duration duration = 0_turns;
+        /**
+         * If true, duration field is ignored and parent intensity is copied.
+         * If true and parent duration was <1, the new effect will not be applied.
+         */
+        bool inherit_duration = false;
+        /** Intensity of the new effect. */
+        int intensity = 0;
+        /** If true, intensity field is ignored and parent effect intensity is copied. */
+        bool inherit_intensity = false;
+
+        bodypart_str_id bp = bodypart_str_id::NULL_ID();
+        bool inherit_body_part = true;
+
+        void load_decay( const JsonObject &obj );
+
+        auto tie() const {
+            return std::tie( type, intensity_requirement, allow_on_decay, allow_on_remove,
+                             duration, inherit_duration, intensity, inherit_intensity );
+        }
+
+        bool operator==( const caused_effect &rhs ) const {
+            return tie() == rhs.tie();
+        }
+    private:
+        void load( const JsonObject &obj );
+};
+
+struct caused_effect_sort_less {
+    public:
+        bool operator()( const caused_effect &lhs, const caused_effect &rhs ) const {
+            if( lhs.type != rhs.type ) {
+                return lhs.type < rhs.type;
+            }
+            return lhs.intensity_requirement < rhs.intensity_requirement;
+        }
+};
```

Comment:
```suggestion
const auto caused_effect_sort_less = []( const caused_effect &lhs, const caused_effect &rhs ) -> bool {
	return lhs.type == rhs.type ? lhs.intensity_requirement < rhs.intensity_requirement : lhs.type < rhs.type;
};
```

this comparator doesn't seem to be used anywhere in PR, would it be intentional?
also, making it lambda might be better for ergonomics.

Reply:
Oh right, I used it for something before, but now it's probably obsolete.